src/lib.rs

```diff
@@ -6,10 +6,12 @@ pub mod icon;
 pub mod iend;
 pub mod ztxt;
 
-use std::io::{Read, Write};
+use std::io::{Cursor, Read, Seek, Write};
 
 /// The PNG magic header
 pub const PNG_HEADER: [u8; 8] = [137, 80, 78, 71, 13, 10, 26, 10];
+pub const IHDR_HEADER: [u8; 8] = [0, 0, 0, 13, 73, 72, 68, 82];
+const ASSUMED_ZTXT_MAX: usize = 500;
```

Comment:
why this number?

Reply:
I looked at a typical DMI file and found that 500 was around the maximum number of bytes until the zTXt header ended. This number is only for performance reasons since it prompts the file reader for how many bytes to try to start by reading before it will need to read another chunk from the file. It's best for it to be smaller to reduce allocations, but large enough to not require most files to read more than once (so no smaller than the length of the file to the end of zTXt, on average)

Reply:
It ultimately has no impact on anything except performance because the reader will read additional bytes from the file if it tries to overread the in-memory buffer